src/Microsoft.DotNet.Cli.Utils/ProjectContextExtensions.cs

```diff
@@ -63,5 +69,118 @@ public static string GetDefaultRootOutputPath(ProjectContext context, string cur
 
             return rootOutputPath;
         }
+        
+        internal static void MakeRunnable(this ProjectContext context, string outputPath, string configuration)
```

Comment:
I'm not a fan of the name here. It doesn't give too many clues about what happens.

From the name I would guess it would mutate the ProjectContext object to represent something runnable. Instead it does a number of operations to mutate the output of a compilation of ProjectContext to make it runnable.

A suggestion: `MakeCompilationOutputRunnable`